src/static/containers/AdminCardAdd/index.js

```diff
@@ -57,59 +56,133 @@ class AdminCardAddView extends Component {
             difficulty_score: 0,
             cost_score: 0,
             help_links: [],
-            stats: {
+            card_stats: {
                 waste_reduction: 0, // in kg/year
                 co2_reduction: 0, // in kg/year
                 water_use_reduction: 0, // in L/year
-                stat_status: 'ACTIVE', // Active/Archived
-                year: 0 // TODO set current year
+                status: 'ACTIVE', // Active/Archived
+                year: (new Date()).getFullYear(),
+                data_sources: []
             },
-            data_source: []
         },
     };
 
+    componentWillMount = () => {
+        const { slug } = this.props.match.params;
+        const { token } = this.props;
+        if (slug) {
+            this.props.actions.cardFetch(token, slug);
+        }
+    };
```

Comment:
Extract everything coming from this.props in order to be readable after: 
const { match, token, actions } = this.props;

Reply:
OK